### PR TITLE
Display the Expiration Date and Number of Allocated Leaves that aren't Carried Over on the Dashboard

### PR DESCRIPTION
**Pull request by abdelrahmankhaledg**
Description of the issue/feature this PR addresses: If you create an allocation with an end date, you will see a warning on the dashboard indicating how long the allocated leaves will be valid before they expire. However, if you create an allocation with an accrual plan having carryover `None` or `Carryover with a maximum`, then some allocated leaves may expire by the carryover date. Yet, there is no warning on the dashboard to indicate the validity period of the allocated leaves that aren't carried over.

This commit fixes the issue by computing the number of days that will expire on the carryover date and displays the expiration date and the number of expired leaves on the dashboard.

task-3978653

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by Bertrand2 — CHANGES_REQUESTED** [review]
Thanks for the PR!
I feel like the code you made for the whole logic is a bit overkill; defining  that much small methods that are called only once is a bit confusing, and I'm guessing we could simplify by looping over the allocations and keeping the one with the closest date, be it date_to or next carryover :thinking: What do you think? :)

**Review by Bertrand2 — CHANGES_REQUESTED** [review]

File: addons/hr_holidays/models/hr_leave_type.py [hunk 1]
```diff
@@ -566,3 +567,49 @@ def get_allocation_data(self, employees, target_date=None):
...
+            accrual_plan_level = allocation._get_current_accrual_plan_level_id(target_date)[0]
+            if expiration_date and expiration_date == closest_expiration_date:
+                expiring_leaves_count += remaining_leaves[allocation]['virtual_remaining_leaves']
+            elif carryover_date and carryover_date == closest_expiration_date:
+                expiring_leaves_count += max(0, remaining_leaves[allocation]['virtual_remaining_leaves'] - accrual_plan_level.postpone_max_days)
+
+        # In case the carryover action is `maximum`, the number of expiring leaves may be zero
+        # which means that no leaves will expire
+        if expiring_leaves_count == 0:
+            closest_expiration_date = False
```
[the remark this answers is not in the document]
Comment (Bertrand2) [reply addons/hr_holidays/models/hr_leave_type.py]: Indeed! My bad I misread your method :) LGTM then

**Review by Bertrand2 — APPROVED** [review]

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@abdelrahmankhaledg Yeah you can move the test to the other task then ^^

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**